linkerd/app/core/src/svc.rs

```diff
@@ -178,14 +178,10 @@ impl<S> Stack<S> {
     /// `ConnectTimeout` error.
     ///
     /// Note that any timeouts errors from the inner service will be wrapped as well.
-    pub fn push_connect_timeout<T>(
+    pub fn push_connect_timeout(
         self,
         timeout: Duration,
-    ) -> Stack<stack::MapErr<tower::timeout::Timeout<S>, impl FnOnce(Error) -> Error + Clone>>
-    where
-        S: Service<T>,
-        S::Error: Into<Error>,
-    {
```

Comment:
These constraints are unnecessary and caused us to have to document `T`. Easier to just remove the constraints.